orderbook/openapi.yml

```diff
@@ -178,6 +178,31 @@ paths:
                 $ref: "#/components/schemas/LegacyFeeInformation"
         404:
           description: sellToken non-existent
+  /api/v1/presignatures:
```

Comment:
Maybe from an API standpoint, it doesn't makes sense to return `signed: false` signatures and just return the current "pre-signature status" for orders/users? Specifically, if the purpose of the query is to:

> The resulto of the presignature query should help the frontend detect when a new order can be created.

Then maybe an endpoint `api/v1/orders/{orderUid}/presigned` that returns either a `true` if there is a valid presignature (i.e. the presignature event for the specified order with the highest `blockNumber` has a value of `signed: true`) and a `false` in case the order UID is not presigned (i.e. there is either no presignature event or the latest event has a value of `signed: false`). Alternatively, it could just return use HTTP status codes `204` and `401` respectively instead of returning data, but this may be slightly abusing HTTP status codes :stuck_out_tongue:.

Reply:
I see. The reason it's built like this is because then one can also see all presignatures for a user. I see however how the usefulness of this might be quite limited: the only use case I can think of is a user who wants to check all presigned orders in order to invalidate them selectively, but in this case it might be reasonable to expect that the user knows the orderUid to be invalidated..

Reply:
> The reason it's built like this is because then one can also see all presignatures for a user

Right, that is a valid usecase. I would still argue that that endpoint `api/v2/presigatures/{user}` could just return a `Vec<OrderUid>` for valid presigatures (i.e. where pre signatures for the order UID with the highest block number has a `true` value for `signed`).

Specifically. Lets say a user does:
```
gpv2_settlement.pre_sign(order_uid_a, true);
// order A expired
gpv2_settlement.pre_sign(order_uid_b, true);
gpv2_settlement.pre_sign(order_uid_b, false);
gpv2_settlement.pre_sign(order_uid_b, false);
gpv2_settlement.pre_sign(order_uid_b, false);
gpv2_settlement.pre_sign(order_uid_b, false);
gpv2_settlement.pre_sign(order_uid_c, true);
gpv2_settlement.pre_sign(order_uid_b, false);
gpv2_settlement.pre_sign(order_uid_c, true);
gpv2_settlement.pre_sign(order_uid_c, true);
```

Then I would argue that:
```
$ curl $BASE_URL/api/v1/presignatures/0x...
[
  "$order_uid_c",
]
```

Would be returned because:
- Order A expired (so the presignature is no longer valid) - although if this were included, it wouldn't be the end of the world. as this would required knowing what the timestamp on the "latest" block is.
- Order B was "unsigned".
- Order C was "unsigned" but then signed again.

Reply:
I agree with B and C but I think A should also be returned nevertheless. If I call `presignatures`, I would not expect that some extra information on the order is used to change what `presigned` means. This is particularly relevant since if we don't have the order associated to the presignature uid then we won't be able to tell if the order is expired, and so it'll appear in the output regardless. So adding a new order (close to the expiration date) would mean that a presigned orders would not be presigned anymore.
The information on validity (validTo + presigned) can be captured by the `order` path.